### PR TITLE
ci: Add Next & SHA tagged container builds

### PR DESCRIPTION
**Pull request by mkuznyetsov**
Signed-off-by: Mykhailo Kuznietsov <mkuznets@redhat.com>

### What does this PR do?
Add new next-build workflow to build Next images (based on last commit) with the intent to replace nightly (daily build)
Nightly build still remains, so it can be disabled later when we will be sure that we don't need it.
Also both nightly & next builds will push a short sha tag for every build, so that it remains preserved

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/19291
https://github.com/eclipse/che/issues/19894

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by mkuznyetsov** [comment]
> LGTM
> 
> Maybe I would have used some `if` instructions in the action to handle if it was run by a scheduler (and then use nightly) or after each commit (and use next) but as the idea is to remove the "nightly job" quickly then it's good to me

Having a new .yml means having a new workflow to display, which would help separating next & nightly runs IMO.
Also it will allow us to have both nightly & next, if some components haven't gone through migration to next yet, and it's critical that they have access to nightly

File: .github/workflows/nightly-build.yml [hunk 1]
```diff
@@ -48,25 +48,39 @@ jobs:
...
       run:
-          ./dockerfiles/build.sh
+          SHORT_SHA1=$(git rev-parse --short HEAD)
+          echo ::set-output name=short_sha1::${SHORT_SHA1}
+          ./dockerfiles/build.sh --tag:nightly --sha-tag
     - name: Push docker images
       run: |
         docker push quay.io/eclipse/che-endpoint-watcher:nightly
+        docker push quay.io/eclipse/che-endpoint-watcher:${{ steps.build.outputs.short_sha1 }}
+
```
Comment (nickboldt) [inline .github/workflows/nightly-build.yml:60]: why build both nightly and next in parallel? seems like a waste of resources and duplication of tags.

File: .github/workflows/next-build.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      run: mvn -B clean install -U -Pintegration
+    - name: Build docker images
+      id: build
+      run:
+          SHORT_SHA1=$(git rev-parse --short HEAD)
+          echo ::set-output name=short_sha1::${SHORT_SHA1}
+          ./dockerfiles/build.sh --tag:next --sha-tag
+    - name: Push docker images
+      run: |
+        docker push quay.io/eclipse/che-endpoint-watcher:next
```
Comment (nickboldt) [inline .github/workflows/next-build.yml:59]: this could be a for loop to reduce it to 4 lines instead of 10

**Review by nickboldt — APPROVED** [review]
As with Florent's comments, as long as the nightly yml goes away soon, I'm +1 here. Would prefer if we remove it now, but I see the value in some continuity until all dependent projects are similarly updated ... are there any that rely on che-server:nightly?

**Review by sparkoo — APPROVED** [review]

File: .github/workflows/next-build.yml (hunk 2 above)
Comment (nickboldt) [reply .github/workflows/next-build.yml:59]:
```
for x in endpoint-watcher keycloak postgres dev server; do
  docker push quay.io/eclipse/che-${x}:next
  docker push quay.io/eclipse/che-${x}:${{ steps.build.outputs.short_sha1 }}
done
```

**Comment by skabashnyuk** [comment]
@mkuznyetsov @nickboldt build is failing. Could you please take a look?

**Pull request merged**